src/clj/puppetlabs/services/jruby/jruby_puppet_core.clj

```diff
@@ -72,16 +72,28 @@
     * :http-client-cipher-suites - A list of legal SSL cipher suites that may
         be used when https client requests are made.
 
+    * :http-client-connect-timeout - The amount of time, in milliseconds, that
+        an outbound HTTP connection will wait for no data to be available before
+        closing the socket. Defaults to 2 minutes. If 0, the timeout is infinite
+        and if negative, the value is undefined (system default).
+
+    * :http-client-socket-timeout - The amount of time, in milliseconds, that an
+        output HTTP connection will wait to connect before giving up. Defaults
```

Comment:
Suggestion: The amount of time, in milliseconds, that an HTTP client will wait to receive data in the response before giving up.

Also, suggest changing this to `:http-client-idle-timeout-milliseconds` to match up with `:idle-timeout-milliseconds`